app/src/main/java/com/babylon/wallet/android/presentation/account/settings/AccountSettingsScreen.kt

```diff
@@ -248,22 +297,113 @@ private fun AccountSettingsContent(
                         .fillMaxWidth()
                         .padding(horizontal = RadixTheme.dimensions.paddingLarge),
                     text = stringResource(R.string.addressAction_showAccountQR),
-                    onClick = onShowQRCodeClick,
+                    onClick = onShowAddressQRCodeClick,
                     enabled = !isXrdLoading
                 )
             }
         }
     }
 }
 
+@Composable
+private fun RenameAccountSheet(
+    modifier: Modifier = Modifier,
+    accountNameChanged: String,
+    onNewAccountNameChange: (String) -> Unit,
+    isNewNameValid: Boolean,
+    onRenameAccountNameClick: () -> Unit,
+    onClose: () -> Unit
+) {
+    Column(
+        modifier = modifier.verticalScroll(rememberScrollState()),
+        verticalArrangement = Arrangement.Center,
+    ) {
+        IconButton(
+            modifier = Modifier.padding(
+                start = RadixTheme.dimensions.paddingXSmall,
+                top = RadixTheme.dimensions.paddingMedium
+            ),
+            onClick = onClose
+        ) {
+            Icon(
+                painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_close),
+                tint = RadixTheme.colors.gray1,
+                contentDescription = null
+            )
+        }
+        Spacer(modifier = Modifier.height(RadixTheme.dimensions.paddingMedium))
+        Text(
+            modifier = Modifier.fillMaxWidth(),
+            text = stringResource(id = R.string.accountSettings_renameAccount_title),
+            style = RadixTheme.typography.title,
+            color = RadixTheme.colors.gray1,
+            textAlign = TextAlign.Center
+        )
+        Spacer(modifier = Modifier.height(RadixTheme.dimensions.paddingMedium))
+        Text(
+            modifier = Modifier.fillMaxWidth(),
+            text = stringResource(id = R.string.accountSettings_renameAccount_subtitle),
+            style = RadixTheme.typography.body1Regular,
+            color = RadixTheme.colors.gray1,
+            textAlign = TextAlign.Center
+        )
+        Spacer(modifier = Modifier.height(RadixTheme.dimensions.paddingDefault))
+        RadixTextField(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = RadixTheme.dimensions.paddingXLarge),
+            onValueChanged = onNewAccountNameChange,
+            value = accountNameChanged,
+            singleLine = true
+        )
+        Spacer(modifier = Modifier.height(RadixTheme.dimensions.paddingXXLarge))
+        RadixPrimaryButton(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = RadixTheme.dimensions.paddingSemiLarge)
+                .padding(bottom = RadixTheme.dimensions.paddingSemiLarge),
+            text = "Update",
```

Comment:
use strings.xml

Reply:
yeah there is not such string at the moment.
And iOS team was not willing to add one when implemented this feature.
I will add one in crowdin